crates/cast/bin/cmd/wallet/mod.rs

```diff
@@ -587,6 +607,72 @@ flag to set your key via:
 
                 sh_println!("{}", success_message.green())?;
             }
+            Self::ChangePassword {
+                account_name,
+                keystore_dir,
+                unsafe_password,
+                unsafe_new_password,
+            } => {
+                // Set up keystore directory
+                let dir = if let Some(path) = keystore_dir {
+                    Path::new(&path).to_path_buf()
+                } else {
+                    Config::foundry_keystores_dir().ok_or_else(|| {
+                        eyre::eyre!("Could not find the default keystore directory.")
+                    })?
+                };
+
+                let keypath = dir.join(&account_name);
+
+                if !keypath.exists() {
+                    eyre::bail!("Keystore file does not exist at {}", keypath.display());
+                }
+
+                let current_password = if let Some(password) = unsafe_password {
+                    password
+                } else {
+                    // if no --unsafe-password was provided read via stdin
+                    rpassword::prompt_password("Enter current password: ")?
+                };
+
+                // decrypt the keystore to verify the current password and get the private key
+                let wallet = PrivateKeySigner::decrypt_keystore(&keypath, current_password.clone())
+                    .map_err(|_| eyre::eyre!("Invalid password - password change cancelled"))?;
+
+                let new_password = if let Some(password) = unsafe_new_password {
+                    password
+                } else {
+                    // if no --unsafe-new-password was provided read via stdin
+                    rpassword::prompt_password("Enter new password: ")?
+                };
+
+                if current_password == new_password {
+                    eyre::bail!("New password cannot be the same as the current password");
+                }
+
+                // Delete the old keystore file
+                std::fs::remove_file(&keypath).wrap_err_with(|| {
+                    format!("Failed to remove old keystore file at {}", keypath.display())
+                })?;
```

Comment:
This should not be necessary as it will simply overwrite the old version, it is also preferred deletion of the previous file is done after the creation of the new one or atomically

https://github.com/roynalnaruto/eth-keystore-rs/blob/85ea8cd5b4dbfcdb3af50e1835540fee83d3b966/src/lib.rs#L256-L258